CHANGELOG.md

```diff
@@ -1,5 +1,73 @@
 # [Versions](https://mui.com/versions/)
 
+## 5.14.14
+
+<!-- generated comparing v5.14.13..master -->
+
+_Oct 17, 2023_
+
+A big thanks to the 23 contributors who made this release possible. Here are some highlights ✨:
```

Comment:
Let's add some highlights, or mention that the release was mostly about bug fixes & docs improvements.